scrapy/utils/conf.py

```diff
@@ -34,6 +34,16 @@ def _map_keys(compdict):
             _check_components(compdict)
             return {convert(k): v for k, v in six.iteritems(compdict)}
 
+    def _ensure_numeric_value(compdict):
+        for name, value in six.iteritems(compdict):
+            if value is None or isinstance(value, (int, float)):
+                continue
+            try:
+                compdict[name] = float(value)
```

Comment:
While we will contibue supporting them in Python 2, order will change, as compared to scrapy 1.3. So it'd be backwards incompatible for users which for whatever reason relied on string priorities (why would anyone do that?). +1 from me to just raise an exception and require users to fix their code.